util/coverage/functest_coverage.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+from pathlib import Path
+from pprint import pformat
+from typing import List
+
+from device_profile_data import extract_profile_data
+from common import (
+    LLD_TARGET,
+    BazelTestType,
+    CoverageParams,
+    run,
+)
+
+# Commands that this script uses
+LLVM_PROFDATA = "llvm-profdata"
+
+
+def handle_libs(device_libs_all: List[str]) -> List[str]:
+    """Filter device libraries that are not compatible with the target.
+
+    Args:
+        device_libs_all: A list of device libraries.
+
+    Returns:
+        `device_libs_all` with incompatible libraries filtered out.
+    """
+    # Remove on_host libs
+    device_libs_incompat = [lib for lib in device_libs_all if "on_host" in lib]
+    logging.info(f"incompatible libraries: {pformat(device_libs_incompat)}")
+    # TODO: may want to add the coverage runtime to avoid undefined symbol warnings
+    return sorted(list(set(device_libs_all) - set(device_libs_incompat)))
+
+
+def handle_objs(merged_library: Path, obj_files: List[str]) -> None:
+    """Create a library from the given object files.
+
+    Args:
+        merged_library: Path where to save the merged library.
+        obj_files: A list of object files.
+    """
+    run(LLD_TARGET, "--warn-unresolved-symbols", "-zmuldefs", "-o",
+        str(merged_library), *obj_files)
+
+
+def handle_test_targets(test_targets: List[str]) -> List[str]:
+    """Choose cw310_test_rom tests from the given list of tests.
+
+    This function also filters wycheproof tests since they take a long time to run.
+
+    Args:
+        test_targets: A list of test targets.
+
+    Returns:
+        cw310_test_rom tests without wycheproof tests.
+    """
+    return [
+        t for t in test_targets
+        if "cw310_test_rom" in t and "wycheproof" not in t
```

Comment:
Is there a way we could filter on test timeouts that are long / eternal? I have a feeling we will add more such tests in in the future that are not named `*_wycheproof_*`.

Reply:
I think so but I filtered these tests just to be able to iterate faster locally. Once we have a stable nightly setup I hope to enable these again.

Reply:
ahh got it. makes sense.